File: msgServer.py
import socket

server = socket.socket(socket.AF_BLUETOOTH, socket.SOCK_STREAM, socket.BTPROTO_RFCOMM)
server.connect(("F8:9E:94:6B:91:02", 4))
server.listen(3)

client, addr = server.accept()

try:
    while True:
        data = client.recv(1024)
        if not data:
            break
        print(f"Message Received on SERVER side: {data.decode('utf-8')}")
        message = input ("ENTER MESSAGE ON SERVER SIDE:")
        client.send(message.encode("utf-8"))
except OSError as e:
    pass

client.close()
server.close()

File: msgServer.js
const BluetoothSerialPort = require('bluetooth-serial-port').BluetoothSerialPort;

const btSerial = new BluetoothSerialPort();
const port = 4; // RFCOMM port number

btSerial.listen(function (clientAddress) {
    console.log(`Client connected from: ${clientAddress}`);

    btSerial.on('data', function (buffer) {
        console.log("Message Received on SERVER side: " + buffer.toString('utf-8'));

        process.stdin.resume();
        process.stdin.setEncoding('utf8');

        process.stdin.on('data', function (message) {
            btSerial.write(new Buffer(message, 'utf-8'), (err, bytesWritten) => {
                if (err) {
                    console.log("Error sending message:", err);
                } else {
                    console.log("Message sent back to client");
                }
            });
        });
    });
});

btSerial.listen(function (clientAddress) {
    console.log(`Server is listening on RFCOMM port ${port}`);
});
